igvc_gazebo/nodes/sim_color_detector/main.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Node responsible for (color) detecting the lines and barrels in the Gazebo
+ * simulator. Receives the raw image and publishes two different B&W images
+ * (one for line detection, one for barrel detection) that each highlight
+ * their respective features in white. Integrated with a ros::Timer to publish at
+ * a specified rate.
+ */
+
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/image_encodings.h>
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/mat.hpp>
+#include <fstream>
+#include <iostream>
+#include <igvc_utils/NodeUtils.hpp>
+
+ros::Publisher img_pub_lines;
+ros::Publisher img_pub_barrels;
+sensor_msgs::Image outmsg_lines;
+sensor_msgs::Image outmsg_barrels;
+
+/*
+ * Responsible for detecting the lines (currently set to a purple color) and
+ * illuminating them as white (and setting all other pixels to black).
+ */
+void handle_image_lines(const sensor_msgs::ImageConstPtr& msg) {
+
+  cv_bridge::CvImagePtr cv_ptr;
+  cv::Mat frame;
+
+  try {
+    cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
+  } catch (cv_bridge::Exception& e) {
+    ROS_ERROR("CV-Bridge error: %s", e.what());
+    return;
+  }
+
+  frame = cv_ptr->image; // Input image
+  cv::Mat output(frame.rows, frame.cols, CV_8UC1, cv::Scalar::all(0)); // Ouput image (B&W)
+
+  // Set color ranges (8 bit BGR) for line detection -> looking for purple
+  const int min_b = 120;
```

Comment:
make all ranges parameters. You might want to create a struct with semantic names for each bounding region.